ts/components/RegionServiceWebView.tsx

```diff
@@ -103,6 +104,12 @@ const RegionServiceWebView: React.FunctionComponent<Props> = (props: Props) => {
       props.handleWebMessage(event.nativeEvent.data);
     }
 
+    const urlParse = new URLParse(props.uri, true);
+
+    if (!event.nativeEvent.url.includes(urlParse.origin)) {
```

Comment:
```suggestion
    if (!event.nativeEvent.url.startsWith(urlParse.origin)) {
```
if we use `includes` and we want to receive messages only from `https://www.google.com` this url passes the check
`https://www.evilsite.com?param=https://www.google.com`


Furthermore I guess this should be the first check of `handleWebviewMessage` function

Reply:
Yes, that's right I'll move it on top!